File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    var l = 0;
    var r = numbers.length-1;
    var res = [];
    var n = numbers[0];
    
    for(var i=0; i<numbers.length-1; i++){
        if((numbers[l]+numbers[r])===target){
            res.push(l+1);
            res.push(r+1);
            return res;
        }
        if((numbers[l]+numbers[r])>target){
            r = r-1;
        }else{
            l=l+1;
        }
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l=0, r=numbers.length-1;
        while(l<r){
            int cur = numbers[l] + numbers[r];
            if(cur == target) return new int[]{l+1, r+1};
            else if(cur>target) r--;
            else l++;
        }
        return new int[]{-1,-1};
    }
}